src/project/manifest/environment.rs

```diff
@@ -33,16 +37,37 @@ impl Borrow<str> for EnvironmentName {
         self.as_str()
     }
 }
+#[derive(Debug, Clone, Error, Diagnostic, PartialEq)]
+#[error("Failed to parse environment name '{attempted_parse}', please use only lowercase letters, numbers and dashes")]
+pub struct ParseEnvironmentNameError {
+    /// The string that was attempted to be parsed.
+    pub attempted_parse: String,
+}
+
+impl FromStr for EnvironmentName {
+    type Err = ParseEnvironmentNameError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let re = Regex::new(r"^[a-z0-9-]+$").unwrap(); // Compile the regex
```

Comment:
This compiles the regex on every invocation. Check lazy_static for a way to avoid that.

Reply:
Done